src/nrnpython/nrnpy_utils.h

```diff
@@ -160,22 +160,6 @@ class PyErr2NRNString {
     char* str_;
 };
 
-
-struct PyLockGIL {
-    PyLockGIL()
-        : state_{PyGILState_Ensure()} {}
-    PyLockGIL(PyLockGIL&&) = delete;
-    PyLockGIL(PyLockGIL const&) = delete;
-    PyLockGIL& operator=(PyLockGIL&&) = delete;
-    PyLockGIL& operator=(PyLockGIL const&) = delete;
-    ~PyLockGIL() {
-        PyGILState_Release(state_);
-    }
-
-  private:
-    PyGILState_STATE state_;
```

Comment:
What's wrong with this class? I suspect it can now be implemented as:
```
class PyLockGIL {
  nanobind::gil_scoped_acquire lock;
};
```

Reply:
Nothing, wrong, only looks useless?

Reply:
As you like, it's not useless, because it was effective at creating a nice protective boundary between NEURON and the ever changing Python binding library.

Reply:
It was not there to create a boundary but a RAII GIL from the C library.